### PR TITLE
fix(language-service): Preserve line endings for diagnostics and template overrides

### LINKED ISSUE #1082: Ivy LS: CRLF line endings breaks sourcemap positions - LS features get provided as if CRLF is 1 character when it is 2
**Describe the bug**

Angular language service does not return completions in interpolations when using CRLF line endings.

**To Reproduce**

1. Generate new angular app: ng new ivy-app
2. Enable the Ivy language service.
3. Change app.component.html content :
```
<div></div>
{{}}
```
4. change file line endings from LF to CRLF
5. move cursor to empty interpolation and ask ide for completions (ctrl+space)

Actual behavior

Completion list is empty. 

**Expected behavior**

Completion list does not depends on template line endings.

**Screenshots**

CRLF

### PR DESCRIPTION
**Pull request by atscott**
This commit addresses the same issue that was fixed in #33241 but this
time for the Ivy compiler integrated Language Service.o

Currently, CRLF line endings are converted to LFs and this causes the
diagnostics span and other LS features to be off in templates that use CRLF.
The line endings must be preserved in order to maintain correct span offset.
The solution is to add an option to the Tokenizer to indicate such preservation.

Fixes angular/vscode-ng-language-service#1082

### CODE REVIEW

File: packages/compiler/src/render3/view/template.ts [hunk 1]
```diff
@@ -2076,14 +2076,17 @@ export interface ParseTemplateOptions {
  * @param options options to modify how the template is parsed
  */
 export function parseTemplate(
-    template: string, templateUrl: string, options: ParseTemplateOptions = {}): ParsedTemplate {
+    template: string, templateUrl: string,
+    options: ParseTemplateOptions&TokenizeOptions = {}): ParsedTemplate {
```
Comment (alxhub) [inline packages/compiler/src/render3/view/template.ts:2080]: We shouldn't expose `TokenizeOptions` here, and instead add `preserveLineEndings` to `parseTemplateOptions`.

**Review by kyliau — COMMENTED** [review]
Please also update https://github.com/angular/angular/blob/be979c907bfe55ecc1a6c90bb449859b4012c207/packages/language-service/ivy/test/legacy/template_target_spec.ts#L30
@JoostK recently changed the parsing options in #40513, so maybe wait for his PR to land first.
The language service part looks good to me, but I'm not sure what is the impact of preserving line endings in the compiler - does it affect compilation in any way?

**Comment by atscott** [comment]
Closing in favor of #40597

**Comment by atscott** [comment]
> Please also update
> 
> https://github.com/angular/angular/blob/be979c907bfe55ecc1a6c90bb449859b4012c207/packages/language-service/ivy/test/legacy/template_target_spec.ts#L30
> 
> 
> @JoostK recently changed the parsing options in #40513, so maybe wait for his PR to land first.
> The language service part looks good to me, but I'm not sure what is the impact of preserving line endings in the compiler - does it affect compilation in any way?

**Comment by kyliau** [comment]
@atscott I think it's still worthwhile to have some basic tests in language service to make sure we don't regress, WDYT?

**Pull request closed without merging**